api/src/main/java/javax/config/spi/ConfigSource.java

```diff
@@ -152,13 +156,48 @@ default int getOrdinal() {
     String getName();
 
     /**
-     * This callback should get invoked if an attribute change got detected inside the ConfigSource.
+     * The callback should get invoked if an attribute change got detected inside the ConfigSource.
      *
-     * @param reportAttributeChange will be set by the {@link javax.config.Config} after this
-     *                              {@code ConfigSource} got created and before any configured values
-     *                              get served.
+     * @param callback will be set by the {@link javax.config.Config} after this
+     *                 {@code ConfigSource} got created and before any configured values
+     *                 get served.
+     * @return ChangeSupport informing the {@link javax.config.Config} implementation about support for changes by this source
+     * @see ChangeSupport
      */
-    default void setOnAttributeChange(Consumer<Set<String>> reportAttributeChange) {
+    default ChangeSupport setOnAttributeChange(Consumer<Set<String>> callback) {
```

Comment:
If you explain to us what the benefit for the application or the ops team is then I'm fine. 
But I fear it simply has no impact and the app cannot do anything anyway. In which case we should leave it out imo.
Convince me ;)

Reply:
We have discussed in person.
This way (with isScannable and ChangeSupport) the Config implementation has all the relevant information about a ConfigSource and can use this information (if needed) to optimize property access (or it can be ignored).